### PR TITLE
Node JS Week 2

### PR DESCRIPTION
**Pull request by hirenkumar91**
Meal Sharing app:

Task Done: Get,Post,Put,Delete implementation on

Reservation and Meal route

### CODE REVIEW

File: api/src/routers/first-meal.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const firstMealroute = express.Router();
+
+firstMealroute.get("/", async (req, res, next) => {
+    console.log("Received request for /api/first-meal");
+    try{
+        const first_meals = await knex("meal").where("id",">",0).orderBy("id","asc").first();
```
Comment (Rassibassi) [inline api/src/routers/first-meal.js]: Shouldn't all meals by definition have `id` larger than `0`?

File: api/src/routers/meals.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const mealRouter = express.Router();
+// get all meal
+mealRouter.get("/", async (req, res, next) => {
+    try{
+        const all_meals = await knex("meal").select("*");
```
Comment (Rassibassi) [inline api/src/routers/meals.js]: `select("*")` is there default, so `knex("meal")` will do

File: api/src/routers/reservations.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    const { id } = req.params;
+    const updatedData = req.body;
+  
+    try {
+        const result = await knex("Reservation")
+            .where({ id })
+            .update(updatedData);
+  
+        if (result) {
+          const updatedReservation = await knex("Reservation").where({ id }).first();
```
Comment (Rassibassi) [inline api/src/routers/reservations.js:49]: you are using `where(...)` in two ways, here in the `put` endpoint and in the above `get` endpoint. Choose one way and be consistent

**Review by Rassibassi — COMMENTED** [review]
Good job see my inline comments

File: api/src/routers/meals.js (hunk 2 above)
Comment (hirenkumar91) [reply api/src/routers/meals.js]: I have made changes now

File: api/src/routers/reservations.js (hunk 3 above)
Comment (hirenkumar91) [reply api/src/routers/reservations.js:49]: changes made as required

File: api/src/routers/first-meal.js (hunk 1 above)
Comment (hirenkumar91) [reply api/src/routers/first-meal.js]: thanks for reivew , made changes accordingly

**Comment by hirenkumar91** [comment]
Dear Mentor , Made all changes as suggested , should i merge

**Pull request merged**